motionplan/tpspace/ptgIK.go

```diff
@@ -114,38 +131,48 @@ func (ptg *ptgIK) MaxDistance() float64 {
 }
 
 func (ptg *ptgIK) Trajectory(alpha, dist float64) ([]*TrajNode, error) {
+	traj := []*TrajNode{}
 	ptg.mu.RLock()
 	precomp := ptg.trajCache[alpha]
 	ptg.mu.RUnlock()
-	if precomp != nil {
-		if precomp[len(precomp)-1].Dist >= dist {
-			// Caching here provides a ~33% speedup to a solve call
-			subTraj := []*TrajNode{}
-			for _, wp := range precomp {
-				if wp.Dist < dist {
-					subTraj = append(subTraj, wp)
-				} else {
-					break
+	if precomp != nil && precomp[len(precomp)-1].Dist >= dist {
+		// Caching here provides a ~33% speedup to a solve call
+		exact := false
+		for _, wp := range precomp {
+			if wp.Dist <= dist {
+				if wp.Dist == dist {
+					exact = true
 				}
+				traj = append(traj, wp)
+			} else {
+				break
 			}
+		}
+		if !exact {
 			time := 0.
-			if len(subTraj) > 0 {
-				time = subTraj[len(subTraj)-1].Time
+			if len(traj) > 0 {
+				time = traj[len(traj)-1].Time
 			}
 			lastNode, err := computePTGNode(ptg, alpha, dist, time)
 			if err != nil {
 				return nil, err
 			}
-			subTraj = append(subTraj, lastNode)
-			return subTraj, nil
+			traj = append(traj, lastNode)
 		}
+	} else {
+		var err error
+		traj, err = ComputePTG(ptg, alpha, dist, defaultResolutionSeconds)
+		if err != nil {
+			return nil, err
+		}
+		ptg.mu.Lock()
+		ptg.trajCache[alpha] = traj
+		ptg.mu.Unlock()
 	}
-	traj, err := ComputePTG(ptg, alpha, dist, defaultResolutionSeconds)
-	if err != nil {
-		return nil, err
-	}
-	ptg.mu.Lock()
-	ptg.trajCache[alpha] = traj
-	ptg.mu.Unlock()
+
 	return traj, nil
 }
+
+func (ptg *ptgIK) Transform(inputs []referenceframe.Input) (spatialmath.Pose, error) {
```

Comment:
Why is this being added?

Reply:
We were previously passing through `Transform` calls to the `PTG`. However, a `PTG` is always 2dof, while the solver frame is able to deal with extended dof.